src/Nethermind/Nethermind.Synchronization/FastBlocks/BodiesSyncFeed.cs

```diff
@@ -72,8 +92,27 @@ public override void InitializeFeed()
                 _barrier = _syncConfig.AncientBodiesBarrierCalc;
                 if (_logger.IsInfo) _logger.Info($"Changed pivot in bodies sync. Now using pivot {_pivotNumber} and barrier {_barrier}");
                 ResetSyncStatusList();
+                if (_blockTree.LowestInsertedBodyNumber is null || _blockTree.LowestInsertedBodyNumber >= _syncConfig.PivotNumberParsed)
```

Comment:
Think about refactoring BodiesSyncFeed and ReceiptSyncFeed to reuse the same (similar) code